packages/common/src/lexicon/lexicon-validation.ts

```diff
@@ -58,7 +58,10 @@ function lexiconToZodSchema(
         zodType = z.number();
         break;
       case 'boolean':
-        zodType = z.boolean();
+        zodType = z.union([
+          z.boolean(),
+          z.string().transform((val) => val === 'true'),
```

Comment:
I don't really know zod, so this might not be a concern. But this reads to me like a boolean value will be valid if it's either a boolean value or a string that says "true". 1. Does this mean a string that says "false" would be considered invalid? 2. The zodType for strings comes before this on line 54. Wouldn't any string, including one that says "true" be caught on this earlier case statement and never reach this boolean case anyway?

Reply:
Good catch, made some fixes. Also made explicit support for 1 & 0